server/service/integration_core_test.go

```diff
@@ -5332,58 +5332,65 @@ func (s *integrationTestSuite) TestPremiumEndpointsWithoutLicense() {
 	createHostAndDeviceToken(t, s.ds, "some-token")
 	s.Do("POST", fmt.Sprintf("/api/v1/fleet/device/%s/migrate_mdm", "some-token"), nil, http.StatusPaymentRequired)
 
+	// software titles
+	// a normal request works fine
+	var resp listSoftwareTitlesResponse
+	s.DoJSON("GET", "/api/latest/fleet/software/titles", listSoftwareTitlesRequest{}, http.StatusOK, &resp)
+	// TODO: there's a race condition that makes this number change from
+	// 0-3, commenting for now since it's not really relevant for this
+	// test (we only care about the response status)
+	// require.NotEmpty(t, 0, resp.Count)
+	// require.Nil(t, resp.SoftwareTitles)
+
+	// a request with a team_id parameter returns a license error
+	resp = listSoftwareTitlesResponse{}
+	s.DoJSON(
+		"GET", "/api/latest/fleet/software/titles",
+		listSoftwareTitlesRequest{}, http.StatusPaymentRequired, &resp,
+		"team_id", "1",
+	)
+}
```

Comment:
Note that I haven't added all that section above, it was already there, it's just the diff generated from moving some code out of this function.